src/hotspot/share/memory/virtualspace.hpp

```diff
@@ -191,6 +196,8 @@ class VirtualSpace {
   size_t _middle_alignment;
   size_t _upper_alignment;
 
+  MEMFLAGS _nmt_flag;
```

Comment:
The VirtualSpace::initialize functions used to initialize these members in the order that they are specified here. That is now messed up by adding the _nmt_flag at the end here, but in the beginning in the initialize function. I would propose that you move it to after _executable, both here and in the initialize function.

Reply:
Fixed.

Reply:
I'd like to see some consistency between ReservedHeap and VritualSpace. Could you change the variables layout and intialization to be:
```
_special
_executable
_nmt_flags
```
in both classes?